### PR TITLE
blog post: Capturing dataset statistics in Apache Spark

### PR DESCRIPTION
**Pull request by pawel-big-lebowski**

### CODE REVIEW

File: website/blog/spark-dataset-statistics/index.mdx [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+---
+
+OpenLineage events enable the creation of a consistent lineage graph, where dataset vertices are connected through the jobs that read from and write to them. This graph becomes even more valuable when its nodes and edges are enriched with additional metadata for practical use cases. One important aspect to capture is the amount of data processed, as it facilitates applications such as cost estimation and data quality monitoring, among others.
+In this post, we introduce recent developments in Spark dataset statistics collection and reporting within OpenLineage events. We outline the basic statistics included, as well as the detailed scan and commit reports generated when processing Iceberg datasets.
+
+<!--truncate-->
+
+## Input and output facets
+
+OpenLineage events are made of three main building blocks: runs, jobs, and datasets. Unfortunately,
```
Comment (wslulciuc) [inline website/blog/spark-dataset-statistics/index.mdx:15]: I would read word:

> Unfortunately, metadata about data processed is both run and dataset property. On one hand, it is a property of a run. On the other hand, it makes sense only in the context of a given dataset.
 

to:

> Metadata about processed data presents an interesting challenge: it is both a property of a run and inherently tied to the context of a specific dataset.

File: website/blog/spark-dataset-statistics/index.mdx [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+We can demonstrate them in action by running Jupyter quickstart demo which emit OpenLineage events to
+docker logs. We start Spark session with:
+```python
+from pyspark.sql import SparkSession
+
+spark = (SparkSession.builder.master('local')
+         .appName('sample_spark')
+         .config('spark.jars.packages', 'org.apache.iceberg:iceberg-spark-runtime-3.5_2.12:1.7.1')
+         .config('spark.jars', '/home/jovyan/openlineage/libs/shadow/openlineage-spark-agent_2.12-1.26.0-SNAPSHOT-shadow.jar')
```
Comment (wslulciuc) [inline website/blog/spark-dataset-statistics/index.mdx:35]: I know the `jar` hasn't been released yet, but maybe add a reference to the (future?) maven artifact vs a local path? 

Or:

```
/path/to/openlineage/libs/shadow/openlineage-spark-agent_2.12-1.26.0-SNAPSHOT-shadow.jar
```

File: website/blog/spark-dataset-statistics/index.mdx [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+    "engine-name": "spark",
+    "iceberg-version": "Apache Iceberg 1.7.1 (commit 4a432839233f2343a9eae8255532f911f06358ef)"
+  },
+  "sequenceNumber": 5
+}
+```
+
+## Conclusion
+
+In this post we introduced the recent developments in Spark dataset statistics collection and reporting within OpenLineage events.
```
Comment (wslulciuc) [inline website/blog/spark-dataset-statistics/index.mdx:197]: I would re-iterate the importance or usage of `scanMetrics` in the conclusion.

**Review by wslulciuc — APPROVED** [review]

File: website/blog/spark-dataset-statistics/index.mdx [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+  .filter("distance >= 150 AND distance < 160") \
+  .write \
+  .insertInto("short_flights")
+```
+which results in a facet:
+
+```json
+{
+  "_producer": "https://github.com/OpenLineage/OpenLineage/tree/1.26.0-SNAPSHOT/integration/spark",
+  "_schemaURL": "https://openlineage.io/spec/facets/1-0-0/IcebergCommitReportInputDatasetFacet.json",
```
Comment (pawel-big-lebowski) [inline website/blog/spark-dataset-statistics/index.mdx:169]: commit report should be output dataset facet, isn't it?

File: website/blog/spark-dataset-statistics/index.mdx (hunk 2 above)
Comment (pawel-big-lebowski) [reply website/blog/spark-dataset-statistics/index.mdx:35]: switching to a released version

**Pull request merged**